ex04.ex

```diff
@@ -75,9 +85,16 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
 
+  def reverse_into_tuple([ evens, odds ] = list), do: { reverse(evens), reverse(odds) }
 
+  def even_odd(list) do
```

Comment:
I have the same comment here about it being a little complex, but I'm not deducting any more.

~~~ elixir
  def even_odd(list) do
    reduce(list, { [], [] }, &partition_even_odd/2)
    |> reverse_two_lists
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}

  defp reverse_two_lists({ evens, odds }) do
    { reverse(evens), reverse(odds) }
  end
~~~